examples/samv/blink/main.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2021, Tomasz Wasilczyk
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <modm/architecture/interface/clock.hpp>
+#include <modm/platform.hpp>
+
+using namespace std::chrono_literals;
+using namespace modm::platform;
+
+struct SystemClock
+{
+	// Chosen to achieve 300MHz system clock
+	static constexpr uint32_t PllAMult = 25;
+	static constexpr uint32_t Frequency = 300'000'000;  // TODO: PllAMult * 12'000'000 (OSC12M);
+
+	static bool inline
+	enable()
+	{
+		ClockGen::setFlashLatency<Frequency>();  // TODO: what is this for?
+
+		// TODO: there seem to be some integer overflow (10ms works fine, 500ms doesn't)
```

Comment:
@salkinium is it possible there is a bug in modm? In the docs (or your article?) I've seen delays of minimum 1s should work - here they clearly don't.

Also, shouldn't modm throw some assertion or fail to compile in such a case?

Reply:
Yes, but only for delay_us. I've specifically written delay_ms to not be limited that way: https://github.com/modm-io/modm/blob/develop/src/modm/platform/core/cortex/delay_impl.hpp.in#L51

Reply:
It does throw an assertion in debug profile and if assertions are used: https://github.com/modm-io/modm/blob/develop/src/modm/platform/core/cortex/delay.cpp.in#L24

Reply:
Actually, I think with 300MHz clock and us_shift of 5, it will overflow around 447ms :(.

Reply:
Ah, yes of course, you need to lower us_shift to 3, since it's bound by max frequency.
The STM32H7 also uses 3, since it goes up to 550MHz.
```
"h7": (4,  4, 3), # CM7  tested on H743 in ITCM
```

Reply:
Fixed. Works as intended for 500ms delays.